.github/workflows/pre-commit.yml

```diff
@@ -8,15 +8,24 @@ on:
       - master
 
 jobs:
+
   pre-commit-checks:
-    runs-on: 'ubuntu-latest'
+    strategy:
+      matrix:
+        os: [ "macos-11", "windows-latest", "ubuntu-18.04", "ubuntu-20.04",  "macos-10.15"]
```

Comment:
We use "macos-11" here is because "macos-latest" is equals to "macos-10.15" instead of "macos-11".